src/main/java/org/springframework/data/jpa/repository/query/Jpa21Utils.java

```diff
@@ -83,17 +86,25 @@ private Jpa21Utils() {
 	 * @see JPA 2.1 Specfication 3.7.4 - Use of Entity Graphs in find and query operations P.117
 	 * @param em must not be {@literal null}.
 	 * @param jpaEntityGraph must not be {@literal null}.
+	 * @param entityType must not be {@literal null}.
 	 * @return the {@link EntityGraph} described by the given {@code entityGraph}.
 	 */
-	public static EntityGraph<?> tryGetFetchGraph(EntityManager em, JpaEntityGraph jpaEntityGraph) {
+	private static EntityGraph<?> tryGetFetchGraph(EntityManager em, JpaEntityGraph jpaEntityGraph, Class<?> entityType) {
 
 		Assert.notNull(em, "EntityManager must not be null!");
 		Assert.notNull(jpaEntityGraph, "EntityGraph must not be null!");
+		Assert.notNull(entityType, "EntityType must not be null!");
 
 		Assert.isTrue(JPA21_AVAILABLE, "The EntityGraph-Feature requires at least a JPA 2.1 persistence provider!");
 		Assert.isTrue(GET_ENTITY_GRAPH_METHOD != null,
 				"It seems that you have the JPA 2.1 API but a JPA 2.0 implementation on the classpath!");
 
-		return em.getEntityGraph(jpaEntityGraph.getName());
+		try {
+			// first check whether an entityGraph with that name is already registered.
+			return em.getEntityGraph(jpaEntityGraph.getName());
+		} catch (Exception ex) {
```

Comment:
There's a few issues I suspect popping up here: the exception being thrown will render the `EntityManager` unusable as defined by the JPA spec (see last paragraph on page 79, JPA 2.1 spec). Thus we either need to create a separate `EntityManager` instance for the check or rather avoid the lookup altogether.

It generally feels a bit wrong to basically register something within in the `EntityManager` in a method that seems to only lookup stuff. Can't we rather change the general model to do the registration once at bootstrap time and remember the name of the entity graph. On an actual method execution we then can look up the graph by name if present or skip this step entirely.